cmd/snap/cmd_snap_op.go

```diff
@@ -622,12 +681,18 @@ func splitSnapAndComponents(name string) (string, []string) {
 	return parts[0], parts[1:]
 }
 
-func snapsAndComponentsFromNames(names []string) ([]string, map[string][]string) {
+func snapsAndComponentsFromNames(names []string, forInstall bool) ([]string, map[string][]string) {
 	snaps := make([]string, 0, len(names))
 	allComps := make(map[string][]string, len(names))
 	for _, name := range names {
 		snap, comps := splitSnapAndComponents(name)
-		snaps = append(snaps, snap)
+		// When installing we implicitly want to install the snap when
+		// we have specified also components, but when removing we
+		// actually want to remove only components if any of them have
+		// been specified.
+		if forInstall || len(comps) == 0 {
+			snaps = append(snaps, snap)
```

Comment:
but this is a bit different, because one uses the /{name} endpoint?

Reply:
This function is called by the *Many methods in fact so we have a list of snaps and a map of components per snap.